src/avram/callbacks.cr

```diff
@@ -5,25 +5,28 @@ module Avram::Callbacks
   # You can set defaults, validate, or perform any other setup necessary for
   # saving.
   #
+  # Optionally you can specify a symbol to `if` or `unless` which allows you to
+  # run this conditionally. The symbol should reference a method you've defined
+  # that returns a truthy/falsey value
+  #
   # ```
   # before_save :run_validations
+  # before_save :run_validations, unless: :admin_bypass?
   #
   # private def run_validations
   #   validate_required name, age
   # end
+  #
+  # private def admin_bypass?
+  #   respond_to?(:user) && user.admin?
+  # end
```

Comment:
```suggestion
  # before_save :validate_can_internet, unless: :too_cool_for_school?
  #
  # private def run_validations
  #   validate_required name, age
  # end
  #
  # private def validate_can_internet
  #   validate_size_of age, min: 13
  # end
  #
  # private def too_cool_for_school?
  #   [true, false].sample
  # end
```

This is probably a bad example, but something simpleish

Reply:
And maybe a more advanced example could go below. Like "Advanced if/unless" that shows using it with field values and passed in `needs`. But that can come later. Not a blocker